data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -103,25 +124,22 @@ public void processUpgrade(String oldVersion, String newVersion) {
 
         News news = newsService.getNewsById(newsTargetsMetadataItem.getObjectId(), false);
         boolean displayed = !news.isArchived() && !StringUtils.equals(news.getPublicationState(), STAGED_STATUS);
-        StringBuilder insertNewsTargetsMetadataItemsPropsQueryStringBuilder = new StringBuilder("INSERT INTO SOC_METADATA_ITEMS_PROPERTIES(METADATA_ITEM_ID, NAME, VALUE) VALUES('");
-        insertNewsTargetsMetadataItemsPropsQueryStringBuilder.append(newsTargetsMetadataItem.getId() + "', '");
-        insertNewsTargetsMetadataItemsPropsQueryStringBuilder.append(NewsUtils.DISPLAYED_STATUS + "', '");
-        insertNewsTargetsMetadataItemsPropsQueryStringBuilder.append(displayed + "')");
-        Query insertNewsTargetsMetadataItemsPropsQuery = entityManager.createNativeQuery(insertNewsTargetsMetadataItemsPropsQueryStringBuilder.toString(), MetadataItemEntity.class);
-        insertNewsTargetsMetadataItemsPropsQuery.executeUpdate();
-        publishedNewsDisplayedPropCount++;
-      }
-      if (transactionStarted && entityManager.getTransaction().isActive()) {
-        entityManager.getTransaction().commit();
-        entityManager.flush();
+        StringBuilder insertNewsTargetsMetadataItemsPropQueryStringBuilder = new StringBuilder("INSERT INTO SOC_METADATA_ITEMS_PROPERTIES(METADATA_ITEM_ID, NAME, VALUE) VALUES('");
+        insertNewsTargetsMetadataItemsPropQueryStringBuilder.append(newsTargetsMetadataItem.getId() + "', '");
+        insertNewsTargetsMetadataItemsPropQueryStringBuilder.append(NewsUtils.DISPLAYED_STATUS + "', '");
+        insertNewsTargetsMetadataItemsPropQueryStringBuilder.append(displayed + "')");
+        Query insertNewsTargetsMetadataItemsPropQuery = entityManager.createNativeQuery(insertNewsTargetsMetadataItemsPropQueryStringBuilder.toString(), MetadataItemEntity.class);
+        insertNewsTargetsMetadataItemsPropQuery.executeUpdate();
+        migratedPublishedNewsCount++;
+        LOG.info("{}/{} published news migrated", migratedPublishedNewsCount, totalPublishedNewsToMigrate);
```

Comment:
this log can be added in for-loop and avoid passing the list size in parameter of method. At the same time, you will be able to log the error/succeeded/total in the message.